web/src/main/java/org/akaza/openclinica/service/UserServiceImpl.java

```diff
@@ -599,12 +599,17 @@ public ParticipantAccessDTO getAccessInfo(String accessToken, String studyOid, S
             logger.error("Participant account not found");
             return null;
         }
-        String accessCode = keycloakClient.getAccessCode(accessToken, pUserAccount.getUserUuid(), customerUuid);
-
-        if (accessCode == null) {
-            logger.error(" Access code from Keycloack returned null ");
-            return null;
+        
```

Comment:
Create a new method "public ParticipantAccessDTO getAccessInfo(String accessToken, String studyOid, String ssid, String customerUuid, UserAccountBean userAccountBean,boolean auditAccessCodeViewing, boolean includeAccessCode)" and use that to control whether to call keycloak to get access code.
Update the original method to call the new method and set "includeAccessCode" to true.